### ISSUE TITLE
Update tutorial to indicate how to pass `requested_scopes` and get non-default tokens

### ISSUE THREAD

**Issue by zonca** [globus/globus-sdk-python#479]
I would like to automate adding users to a group.

I need to create an authorizer with the right permissions to create a `GroupClient`.

I gave my app the scope `urn:globus:auth:scope:groups.api.globus.org:all`, however when I get back the response token, I see the other scopes fine:

```
In [12]: token_response.by_scopes["urn:globus:auth:scope:transfer.api.globus.org:all"]
Out[12]: 
{'scope': 'urn:globus:auth:scope:transfer.api.globus.org:all',
 'access_token': 'xxxx',
 'refresh_token': 'xxxxxx',
 'token_type': 'Bearer',
 'expires_at_seconds': 1632622894,
 'resource_server': 'transfer.api.globus.org'}
```

But if I try with:

    token_response.by_scopes["urn:globus:auth:scope:groups.api.globus.org:all"]

I get:

    KeyError: 'Scope specifier "urn:globus:auth:scope:groups.api.globus.org:all" contains scope "urn:globus:auth:scope:groups.api.globus.org:all" which was not found'

**Comment by sirosen** [comment]
> I gave my app the scope `urn:globus:auth:scope:groups.api.globus.org:all`

Do you mean that you passed this scope string for `requested_scopes` when starting the oauth flow, or that you set it in the web interface when creating your client?
As far as I know, the setting in the web UI is a legacy feature and doesn't do anything in the normal case. (We should remove it from the SDK tutorial.)

You should be able to pass that scope in when starting the flow. See [docs on oauth2_start_flow parameters](https://globus-sdk-python.readthedocs.io/en/stable/services/auth.html#globus_sdk.NativeAppAuthClient.oauth2_start_flow).

Also, as of SDK v3.0 you shouldn't need to hardcode the scope string. You can get it from `globus_sdk.scopes` under the name `GroupsScopes` or as `GroupsClient.scopes` (see [doc on the scopes and scope builders](https://globus-sdk-python.readthedocs.io/en/stable/scopes.html)). e.g.
```python
from globus_sdk import GroupsClient

auth_client = ... # create an auth client for your app, varies by app type

auth_client.oauth2_start_flow(requested_scopes=[GroupsClient.scopes.all])

# follow the flow and get a token response, again this varies a little by app type
...
token_response = ...

# either using the scope
tokens = token_response.by_scopes[GroupsClient.scopes.all]
# or using the resource server
tokens = token_response.by_resource_server[GroupsClient.scopes.resource_server]
```

I generally would prefer the `by_resource_server` usage because that attribute is just a dictionary and you can easily inspect it. `by_scopes` is a fancy getter which crawls that same dictionary.

**Comment by zonca** [comment]
thanks @sirosen
it works great now, please add this to the docs!

**Comment by sirosen** [comment]
I've retitled this to be an "action item" which we can do and close. Just so that we have a clearer idea of what it means to close this.

I think the appropriate fix is for the tutorial doc to explicitly call out the `requested_scopes` parameter. An example flow specifically for Groups would probably be a good choice, as I expect it will be a common case.

We've had some improvements to the tutorial in the backlog for ages. Maybe it's finally time that we get to some of them...

**Comment by zonca** [comment]
@sirosen I wrote a small tutorial and shared my scripts in case it comes useful for your docs: https://zonca.dev/2021/09/globus-groups-python-sdk.html

**Comment by sirosen** [comment]
I just noticed that this was still open. It should have been closed back in #511, when I did a pass over the tutorial to do some updates.
